### PR TITLE
#84 adding `Replace()`

### PR DESCRIPTION
**Pull request by grundprinzip**
### What changes were proposed in this pull request?
Adding support for the basic version of `Replace()`

### Why are the changes needed?
Compatibility

### Does this PR introduce _any_ user-facing change?
Adding new functions

### How was this patch tested?
Added IT

### CODE REVIEW

File: spark/sql/dataframe.go [hunk 1]
```diff
@@ -146,6 +146,19 @@ type DataFrame interface {
...
+	// to the type of the existing column.
+	//
+	// For numeric replacements all values to be replaced should have unique
+	// floating point representation.
+	Replace(ctx context.Context, toReplace []types.PrimitiveTypeLiteral,
+		values []types.PrimitiveTypeLiteral) (DataFrame, error)
+	// ReplaceWithColumns is an alias for Replace but allows to specify a subset of columns that
+	// should be used for the replacement.
+	ReplaceWithColumns(ctx context.Context, toReplace []types.PrimitiveTypeLiteral,
+		values []types.PrimitiveTypeLiteral, cols []string) (DataFrame, error)
```
Comment (alexott) [inline spark/sql/dataframe.go:161]: if we allow to use empty `cols`, then maybe we can use `cols ...string` instead, so people can simply omit the value?

**Review by alexott — APPROVED** [review]
just small comment

**Review by HyukjinKwon — APPROVED** [review]

File: spark/sql/dataframe.go (hunk 1 above)
Comment (grundprinzip) [reply spark/sql/dataframe.go:161]: will do

**Comment by grundprinzip** [comment]
merging to master

**Pull request closed without merging**